base/shell/explorer/traywnd.cpp

```diff
@@ -99,6 +99,35 @@ VOID RestoreWindowPos()
     g_WindowPosBackup.RemoveAll();
 }
 
+BOOL CanDialogSysMinimize(HWND hwnd)
+{
+    if (::GetClassLongPtrW(hwnd, GCW_ATOM) != reinterpret_cast<ULONG_PTR>(WC_DIALOG))
+        return FALSE;
+    return (::IsWindowVisible(hwnd) && !::IsIconic(hwnd) && ::IsWindowEnabled(hwnd));
+}
+
+BOOL ShouldMinimize(HWND hwnd)
+{
+    if (::IsWindowVisible(hwnd) && !::IsIconic(hwnd) && ::IsWindowEnabled(hwnd))
+    {
+        DWORD style = static_cast<DWORD>(::GetWindowLongPtrW(hwnd, GWL_STYLE));
+        if ((style & WS_MINIMIZEBOX) && (style & WS_CAPTION) == WS_CAPTION)
+        {
+            if (style & WS_SYSMENU)
+            {
+                HMENU hSysMenu = ::GetSystemMenu(hwnd, FALSE);
+                if (hSysMenu)
+                    return !(::GetMenuState(hSysMenu, SC_MINIMIZE, MF_BYCOMMAND) & MF_DISABLED);
```

Comment:
Does your interactive test app also show that this is a condition that needs to be respected?